### ISSUE TITLE
tfjs converter from pb file removed, how to transform graph now?

### ISSUE THREAD

**Issue by startupgurukul** [tensorflow/tfjs#1379]
Hi folks,  until now, one of the processes of creating a tf.js model has been to use the frozen pb file, prune unnecessary nodes with TF  graph_transform tool and then convert to tf.js.

Now that converting from pb is not supported in tf.js converter from 1.0(other than the pb2json way), can you enlighten us on how to accomplish the same now with 1.x? I know we can do the same, post-conversion by trimming the layers, want to know if we can do the same during conversion, maybe tensorflowjs_converter can take two optional params input_nodes and output_nodes

**Comment by pyu10055** [comment]
@startupgurukul there are two ways to convert the frozen pb to work with the new loadGraphModel API:

1. If you have converted the frozen pb file previously, you can use the pb2json script to convert it to JSON format. details is in the FAQ section of the README file. 

```
git clone git@github.com:tensorflow/tfjs-converter.git
cd tfjs-converter
yarn
yarn ts-node tools/pb2json_converter.ts pb_model_directory/ json_model_directory/
pb_model_directory is the directory where the model generated by previous version is located. json_model_directory is the destination directory for the converted model.
```

2. To convert the frozen pb file from scratch, you need to use the 0.8.0 tensorflowjs pip, and add --otuput-json=true to the command line.
You can find the details [here](https://github.com/tensorflow/tfjs-converter/blob/0.8.x/README.json.md).

**Comment by startupgurukul** [comment]
I kind of knew about both of them but thank you for the detailed steps, let me rephrase my question how do we graph pruning while using the tfjs 1.x converter, some way of providing two optional params             --input_arraysand --input_arrays

something similar to tflite converter 
https://www.tensorflow.org/lite/convert/cmdline_examples

**Comment by pyu10055** [comment]
@startupgurukul in tfjs 1.x converter the graph pruning is done automatically for SavedModel, since SavedModel has signatureDef that specifies the input/output nodes.

For converting frozen graph models, the best option is to use 0.15.x converter with --output_json flag, it generates 1.x compatible JSON web models.

**Comment by startupgurukul** [comment]
ok thank you
